pdfplumber/utils.py

```diff
@@ -494,7 +484,7 @@ def clip_obj(obj, bbox):
     for attr in ["x0", "top", "x1", "bottom"]:
         copy[attr] = dims[attr]
 
-    if dims["top"] != obj["bottom"] or dims["top"] != obj["bottom"]:
```

Comment:
I will be tweaking this further, but thank you for pointing out this redundancy.